File: framework/ecommerce-framework-api/src/main/java/com/sdl/ecommerce/api/model/QuerySuggestion.java
package com.sdl.ecommerce.api.model;

/**
 * Query Suggestion
 * Gives a suggestion on mistyped search phrase.
 *
 * @author nic
 */
public interface QuerySuggestion {

    /**
     * @return original search phrase
     */
    String getOriginal();

    /**
     * @return suggestion on search phrase
     */
    String getSuggestion();

    /**
     * @return estimated results for current suggestion. Returns NULL if there is no estimates available
     */
    Integer getEstimatedResults();

}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.Rest/Model/QuerySuggestion.cs
using SDL.ECommerce.Api.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SDL.ECommerce.Rest.Model
{
    class QuerySuggestion : IQuerySuggestion
    {
        public string Original { get; set; }

        public string Suggestion { get; set; }

        public int? EstimatedResults { get; set; }
    }
}
